### PR TITLE
flake8

### PR DESCRIPTION
**Pull request by m-kuhn**
## Description
Runs flake8, because things like https://github.com/qgis/QGIS/pull/36214#pullrequestreview-406054151 shouldn't go unnoticed.

### CODE REVIEW

File: .github/workflows/pyflakes.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+on: [push, pull_request]
+
+jobs:
+  flake8_py3:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup Python
+        uses: actions/setup-python@v1
+        with:
+          ython-version: 3.7.6
```
Comment (nyalldawson) [inline .github/workflows/pyflakes.yml:12]: Typo?

File: python/plugins/processing/algs/qgis/RasterCalculator.py [hunk 2]
```diff
@@ -197,7 +197,7 @@ def processBeforeAddingToModeler(self, algorithm, model):
         for i in list(model.inputs.values()):
             param = i.param
             if isinstance(param, QgsProcessingParameterRasterLayer) and "{}@".format(param.name) in expression:
-                values.append(ValueFromInput(param.name()))
+                values.append(ValueFromInput(param.name()))  # noqa TODO: how is that supposed to work??
```
Comment (m-kuhn) [inline python/plugins/processing/algs/qgis/RasterCalculator.py:200]: @nyalldawson do you have an idea what "ValueFromInput" is supposed to call or do? It seems to originate in some ModellerAlgorithm python module that no longer exists. This is the only leftover from this.

File: python/plugins/processing/algs/qgis/RasterCalculator.py [hunk 3]
```diff
@@ -208,7 +208,7 @@ def processBeforeAddingToModeler(self, algorithm, model):
                 for out in alg.algorithm.outputs:
                     if (isinstance(out, QgsProcessingOutputRasterLayer) and
                             "{}:{}@".format(alg.modeler_name, out.name) in expression):
-                        values.append(ValueFromOutput(alg.modeler_name, out.name))
+                        values.append(ValueFromOutput(alg.modeler_name, out.name))  # noqa TODO: how is that supposed to work??
```
Comment (m-kuhn) [inline python/plugins/processing/algs/qgis/RasterCalculator.py:211]: @nyalldawson Same here.

File: python/plugins/processing/algs/qgis/RasterCalculator.py (hunk 2 above)
Comment (nyalldawson) [reply python/plugins/processing/algs/qgis/RasterCalculator.py:200]: Kill it. processBeforeAddingToModeler is not used anywhere

File: tests/src/python/test_qgsserver_wfst.py [hunk 4]
```diff
@@ -92,7 +92,7 @@ def setUpClass(cls):
         cls.server = subprocess.Popen([sys.executable, server_path],
                                       env=os.environ, stdout=subprocess.PIPE)
         line = cls.server.stdout.readline()
-        cls.port = int(re.findall(b':(\d+)', line)[0])
+        cls.port = int(re.findall(r':(\d+)', line)[0])
```
Comment (m-kuhn) [inline tests/src/python/test_qgsserver_wfst.py:95]:
```suggestion
        cls.port = int(re.findall(br':(\d+)', line)[0])
```

**Comment by nyalldawson** [comment]
Nice work @m-kuhn!

Looking at this I think we should consider moving more of the layout checks to a similar approach (e.g the shell check and license checks)

File: .ci/travis/scripts/ctest2travis.py [hunk 5]
```diff
@@ -83,7 +83,7 @@ def start_test_fold():
         end_fold() # tag=build
         start_test_fold()
 
-    if re.search('\*\*\*Failed', updated_line) or re.search('\*\*\*Timeout', updated_line):
+    if re.search('***Failed', updated_line) or re.search('***Timeout', updated_line):
```
Comment (3nids) [inline .ci/travis/scripts/ctest2travis.py:86]: is this correct? It looks like it breaks the regex

Comment (3nids) [inline .ci/travis/scripts/ctest2travis.py:86]:
```suggestion
    if re.search(r'\*\*\*Failed', updated_line) or re.search(r'\*\*\*Timeout', updated_line):
```

File: scripts/qgis_fixes/fix_signals.py [hunk 6]
```diff
@@ -59,7 +59,7 @@ class FixSignals(fixer_base.BaseFix):
 
     def transform(self, node, results):
         signal = results.get("signal").value
-        signal = re.sub('^["\']([^(]+)(?:\(.*\))?["\']$', '\\1', signal)
+        signal = re.sub('^["\']([^(]+)(?:(.*))?["\']$', '\\1', signal)
```
Comment (m-kuhn) [inline scripts/qgis_fixes/fix_signals.py:62]:
```suggestion
        signal = re.sub('^["\']([^(]+)(?:\\(.*\\))?["\']$', '\\1', signal)
```

File: tests/src/python/test_qgsserver.py [hunk 7]
```diff
@@ -51,9 +51,9 @@
 
 
 # Strip path and content length because path may vary
-RE_STRIP_UNCHECKABLE = b'MAP=[^"]+|Content-Length: \d+'
-RE_ELEMENT = b'</*([^>\[\s]+)[ >]'
-RE_ELEMENT_CONTENT = b'<[^>\[]+>(.+)</[^>\[\s]+>'
+RE_STRIP_UNCHECKABLE = rb'MAP=[^"]+|Content-Length: \d+'
```
Comment (m-kuhn) [inline tests/src/python/test_qgsserver.py:54]:
```suggestion
RE_STRIP_UNCHECKABLE = br'MAP=[^"]+|Content-Length: \d+'
```

File: .ci/travis/scripts/ctest2travis.py [hunk 8]
```diff
@@ -97,14 +97,14 @@ def start_test_fold():
                 in_failure = False
             else:
                 updated_line = colored(updated_line, 'yellow')
-        elif re.search('\*\*\* Segmentation fault', updated_line):
+        elif re.search('*** Segmentation fault', updated_line):
```
Comment (m-kuhn) [inline .ci/travis/scripts/ctest2travis.py:100]:
```suggestion
        elif re.search(r'\*\*\* Segmentation fault', updated_line):
```

File: .ci/travis/scripts/ctest2travis.py [hunk 9]
```diff
@@ -97,14 +97,14 @@ def start_test_fold():
...
             start_fold('segfault')
             updated_line = colored(updated_line, 'magenta')
         elif re.match('  Test failed: Segmentation fault', updated_line):
             end_fold()
 
         else:
-            if re.match('(FAIL|ERROR)[:\!].*', updated_line):
+            if re.match('(FAIL|ERROR)[:!].*', updated_line):
```
Comment (m-kuhn) [inline .ci/travis/scripts/ctest2travis.py:107]:
```suggestion
            if re.match(r'(FAIL|ERROR)[:\!].*', updated_line):
```

**Comment by m-kuhn** [comment]
> Looking at this I think we should consider moving more of the layout checks to a similar approach (e.g the shell check and license checks)

Yes, that would give a much quicker feedback

**Pull request merged**